org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/EclipseResourceFileSystemAccess2.java

```diff
@@ -248,9 +248,11 @@ protected void generateFile(IFile file, InputStream content, IFile traceFile, Ch
 			if (file.exists()) {
 				if (outputConfig.isOverrideExistingResources()) {
 					if (hasContentsChanged(file, content)) {
```

Comment:
unfortunately `hasContentsChanged()` also reads the stream, so after `hasContentsChanged()` the streams's curser is probably at the stream's end. I think for streams without marker support we'd need to read the stream into a buffer once and then use that buffer's contents for `hasContentsChanged()` and `file.setContents`.